charts/astronomer/templates/_helpers.yaml

```diff
@@ -41,6 +41,37 @@
 {{- end }}
 {{- end }}
 
+{{/*
+Returns a repository prefix to be used to construct image names throughout
+the platform. Defaults to quay.io/astronomer but uses the value defined in
+global.privateRegistry.repository when enabled.
+*/}}
+{{ define "astronomer.platform_registry" -}}
+{{- if .Values.global.privateRegistry.enabled -}}
+{{ .Values.global.privateRegistry.repository }}
+{{- else -}}
+quay.io/astronomer
+{{- end }}
+{{- end }}
+
+{{/*
+Repository that holds the authSidecar image (does not include tag)
+*/}}
+{{ define "authSidecar.repository" -}}
+{{- if .Values.global.authSidecar.repository -}}
+{{ .Values.global.authSidecar.repository }}
+{{- else -}}
+{{ template "astronomer.platform_registry" . }}/ap-auth-sidecar
+{{- end }}
+{{- end }}
+
+{{/*
+Image of the authSidecar image (same as authSidecar.repository but with tag)
+*/}}
+{{ define "authSidecar.image" -}}
+{{ template "authSidecar.repository" . }}:{{ .Values.global.authSidecar.tag }}
+{{- end }}
```

Comment:
Why does this section need to be different from all of the other privateRegistry image templates?

Reply:
The templates we are currently using break image name over-riding when when privateRegistry is enabled.

Reply:
Can you be more specific? Maybe it's something we should fix.